### ISSUE TITLE
Using the Mono Track for a 7 channel measurement

### ISSUE THREAD

**Issue by rpx91** [jaakkopasanen/Impulcifer#19]
I tried to record a full 7 channel setup today and ran into an error: 

> (venv) C:\Python\Impulcifer-master>python impulcifer.py --dir_path=data\my_hrir --test_signal=data\sweep-6.15s-48000Hz-32bit-2.93Hz-24000Hz.wav --speakers=C,FL,FR,SL,SR,RL,RR --compensate_headphones
> Traceback (most recent call last):
>   File "impulcifer.py", line 167, in <module>
>     main(**create_cli())
>   File "impulcifer.py", line 33, in main
>     hrir.open_recording(recording, speakers=speakers)
>   File "C:\Python\Impulcifer-master\hrir.py", line 54, in open_recording
>     speaker = speakers[i + j]
> IndexError: list index out of range

I used the mono recording (sweep-seg-FL-mono-6.15s-48000Hz-32bit-2.93Hz-24000Hz) and switched the balance to the right - that suited my setup . Sat on a swivel chair and sat through the various positions.

Tried to run the command

`(python impulcifer.py --dir_path=data\my_hrir --test_signal=data\sweep-6.15s-48000Hz-32bit-2.93Hz-24000Hz.wav --speakers=C,FL,FR,SL,SR,RL,RR --compensate_headphones)`

But the above error on pastebin occured.

I uploaded my recording.wav [here](https://drive.google.com/file/d/1fcRUQnynGhVvP4-P7_KXZfDk6-Bw6Xtm/view?usp=sharing): 

I tried a bunch of things to no avail - splitting stereo tracks to mono etc.

The reason I used the mono track is I can't understand how to get a proper recording from the stereo sweep. You just can't do the side surrounds or center channel that way.

**Comment by rpx91** [comment]
Turns out this isn't a mono track issue.

I tried it with the stereo track and it's getting the same error. Soon as I remove the center channel from recording.wav the error goes.

Is this because Impulficier is expecting an even number of channels? How do you record the center channel?

Although I avoided the loop count error I got a new error following your spinorama stereo speakers technique. I did incorrectly have the gain set on one of the channels to lower than the other, but I thought Impulficier corrects that anyway.

```
(venv) C:\Users\dushy\Impulcifer>python impulcifer.py --dir_path=data\my_hrir --test_signal=data\sweep-6.15s-48000Hz-32bit-2.93Hz-24000Hz.wav --speakers=FL,FR,SR,BR,BL,SL --compensate_headphones
Traceback (most recent call last):
  File "impulcifer.py", line 167, in <module>
    main(**create_cli())
  File "impulcifer.py", line 43, in main
    hrir.crop_heads()
  File "C:\Users\dushy\Impulcifer\hrir.py", line 161, in crop_heads
    raise ValueError(speaker + ' impulse response has lower delay to right ear than to left.')
ValueError: FL impulse response has lower delay to right ear than to left.
```

**Comment by rpx91** [comment]
I can confirm the process works fine for the stereo sweep. I was having a lot of difficulty understanding how the center was working - but then I saw earlier in the guide that you just use X to ignore the measurement of a non existent speaker location. I was trying to delete it from recording.wav which was resulting in errors.

**Comment by jaakkopasanen** [comment]
Thanks a lot for reporting this and for providing the measurement file. There was a bug indeed and I fixed it. Now at least the single speaker measurement you provided and the stereo speaker measurements I have made work. I suspect surround speaker setup measurements should work also.

It's true that the stereo measurement process can be quite confusing and the guide doesn't perhaps explain it as well as it should. I need to create an animation which explains the speaker positions and/or upload a video about the measurement process to Youtube.

**Comment by rpx91** [comment]
Excellent thanks! Working now.

Turns out I got a BL lower delay to right than left again. Must have made an error in that measurement too.

The only reason it was confusing was due to not understanding that there's an X to ignore the measurement for the center channel. Your stereo method actually results in bang on placement - doing it mono is hard to get it exactly right without a laser protractor. I'd say it only needs an animation.
